File: app/src/main/java/com/amicly/alamofoursquare/data/remote/FourSquareService.java
package com.amicly.alamofoursquare.data.remote;

import com.amicly.alamofoursquare.model.photo.PhotoSearchResult;
import com.amicly.alamofoursquare.model.venue.VenueSearchResult;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface FourSquareService {

    String VENUES = "venues";
    String SEARCH_NEAR_AUSTIN = VENUES + "/search?near=Austin,TX";
    String PHOTOS = VENUES + "/{venueId}/photos";

    @GET(SEARCH_NEAR_AUSTIN)
    Single<VenueSearchResult> searchEvents(@Query("query") String searchQuery);

    @GET(PHOTOS)
    Single<PhotoSearchResult> getPhotos(@Path("venueId") String venueId);
}

File: app/src/main/java/com/amicly/alamofoursquare/data/remote/FourSquareService.kt
package com.amicly.alamofoursquare.data.remote

import com.amicly.alamofoursquare.model.photo.PhotoSearchResult
import com.amicly.alamofoursquare.model.venue.VenueSearchResult

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface FourSquareService {

    @GET(SEARCH_NEAR_AUSTIN)
    fun searchEvents(@Query("query") searchQuery: String): Single<VenueSearchResult>

    @GET(PHOTOS)
    fun getPhotos(@Path("venueId") venueId: String): Single<PhotoSearchResult>

    companion object {

        const val VENUES = "venues"
        const val SEARCH_NEAR_AUSTIN = "$VENUES/search?near=Austin,TX"
        const val PHOTOS = "$VENUES/{venueId}/photos"
    }
}
